### ISSUE TITLE
Fix or delete tests that make invalid assumptions (about codegen or anything else)

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#5081]
Questionable tests can often be identified by their existence in exclusion files under the tests folder or by tags such as <JitOptimizationSensitive> in their project files.  See, for example, #5038.

**Comment by yizhang82** [comment]
@RussKeldorph is #5038 the right issue you want to refer to?

**Comment by RussKeldorph** [comment]
@yizhang82 No.  I corrected the description.  Thank you for pointing it out.
